GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs

```diff
@@ -977,18 +1031,25 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
                     {
                         foreach (EffectEvent effect in zhaitanPoisonImpactEffects)
                         {
-                            int end = (int)effect.Time;
-                            int start = end - 2000;
-                            replay.AddDecorationWithGrowing(new CircleDecoration(200, (start, end), "rgba(49, 71, 0, 0.2)", new PositionConnector(effect.Position)), end);
+                            (long start, long end) lifespan = (effect.Time - 2000, effect.Time);
+                            replay.AddDecorationWithGrowing(new CircleDecoration(200, lifespan, Colors.LightMilitaryGreen, 0.2, new PositionConnector(effect.Position)), lifespan.end);
                         }
                     }
                     if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.HarvestTempleZhaitanPutridDelugeAoE, out IReadOnlyList<EffectEvent> zhaitanPoisonAoeEffects))
                     {
                         foreach (EffectEvent effect in zhaitanPoisonAoeEffects)
                         {
-                            int start = (int)effect.Time;
-                            int end = (int)Math.Min(target.LastAware, start + 10000);
-                            replay.AddDecorationWithBorder(new CircleDecoration(200, (start, end), "rgba(49, 71, 0, 0.2)", new PositionConnector(effect.Position)), "rgba(200, 0, 0, 0.5)");
+                            int fieldDuration = 10000;
+                            (long start, long end) lifespan = (effect.Time, effect.Time + fieldDuration);
+                            if (lifespan.start > target.LastAware)
+                            {
+                                lifespan.end = effect.Time + 10000;
```

Comment:
We only need the else no? Also those + 10000 should be + fieldDuration

Reply:
Yeah can invert the if to be lifespan.Start < target.LastAware and only have the content of the else